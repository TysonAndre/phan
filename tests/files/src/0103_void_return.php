<?php

/** @return null */
function f0() { return null; }

/** @return null */
function f1() { return 42; }

/** @return null */
function f2() {}

/** @return void - doesn't warn because null can currently cast to void and callers can't tell. */
function g0() { return null; }

/** @return void */
function g1() { return 42; }

/** @return void */
function g2() {}
