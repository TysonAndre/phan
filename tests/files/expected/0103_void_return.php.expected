%s:7 PhanTypeMismatchReturnProbablyReal Returning 42 of type 42 but f1() is declared to return null (no real type) (the inferred real return type has nothing in common with the declared phpdoc return type)
%s:16 PhanTypeMismatchReturnProbablyReal Returning 42 of type 42 but g1() is declared to return void (no real type) (the inferred real return type has nothing in common with the declared phpdoc return type)
